src/app/pages/office/cat-b/office.cat-b.page.ts

```diff
@@ -1,58 +1,341 @@
+
   }
 
-  ngOnInit() {
+  ngOnInit(): void {
     super.onInitialisation();
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+    const category$ = currentTest$.pipe(
+      select(getTestCategory),
+    );
     this.pageState = {
       ...this.commonPageState,
+      displayShowMeQuestion$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+          select(getVehicleChecks),
+          select(getShowMeQuestion),
+        )),
+        map(([outcome, question]) =>
+          this.outcomeBehaviourProvider.isVisible(outcome, 'showMeQuestion', question)),
+      ),
+      displayTellMeQuestion$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+          select(getVehicleChecks),
+          select(getTellMeQuestion),
+        )),
+        map(([outcome, question]) =>
+          this.outcomeBehaviourProvider.isVisible(outcome, 'tellMeQuestion', question)),
+      ),
+      displayDrivingFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDrivingFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      displaySeriousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getSeriousFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      displayDangerousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDangerousFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      showMeQuestion$: currentTest$.pipe(
+        select(getTestData),
+        select(getVehicleChecks),
+        select(getShowMeQuestion),
+      ),
+      showMeQuestionOptions$: currentTest$.pipe(
+        select(getTestOutcome),
+        map((outcome) =>
+          getShowMeQuestionOptions(this.showMeQuestions, outcome, this.outcomeBehaviourProvider)),
+      ),
+      tellMeQuestionText$: currentTest$.pipe(
+        select(getTestData),
+        select(getVehicleChecks),
+        select(getSelectedTellMeQuestionText),
+      ),
+      etaFaults$: currentTest$.pipe(
```

Comment:
This seems common between most if not all categories, would suggest moving to base page

Reply:
comments below